### DISCUSSION TITLE
Executing match only when it's the first thing in a new terminal prompt

### DISCUSSION THREAD

**Discussion by fharper** [espanso/espanso#1724] [Q&A]
I have a lot of triggers I use only in the terminal with a `filter_class`, but some of them can also be expended in another command I ran caused the abbreviation I use is detected. As an example, I have the trigger `bs` which could be expanded when I don't want to if I use a command, let's say, that is `tabs` since the letters `bs` is in `tabs`. You get the point...

To solve this issue, I use a regex instead of a trigger. It is working to prevent the case mentioned above, but it doesn't always work. Sometimes Espanso doesn't detect it, and I can't put my hand on what is preventing it on some new prompt in the Terminal (I'm using iTerm2). Actually, I say Espanso, but it may just be anything on the prompt that can prevent the regex to be valid...

I was wondering if it was about an exit code from the previous command, some invisible characters, my prompt characters or shell configuration (Zsh with OMZ) or even if my regex wasn't valid (using `[^ a-z]` before the abbreviation I want to use). I tried multiple regex variations that could achieve the same goal, and it's the simplest I came up with which is working in most cases. 

When the issue happens, I just `CTRL + C` to cancel the actual terminal prompt, and move to a new one. See for yourself

https://github.com/espanso/espanso/assets/1170989/1f539345-8aff-47f2-b7b9-d8b5761e05db

Any idea what is happening, and how to fix that? It's been a while I wanted to ask, so I'm starting to get used to CTRL+C often, but would really love to solve this problem :)

**Comment by smeech** [comment]
Does the `word: true` property not fix this? See https://espanso.org/docs/matches/basics/#word-triggers.
It's part of the reason many of us prefix our triggers with ":" or ";" to avoid the problem.

**Reply by fharper** [reply]
Thanks for answering, I appreciate!

As mentioned in the documentation you pointed, it only works if the trigger is surrounded by word separators (such as spaces, commas and newlines) which is not the intended behavior I want.

I have a lot of other triggers starting with `:` as you mentioned, but I'm using those triggers for years with another application, so I would highly prefer to find how to do that basic triggering with Espanso than changing 30+ shortcuts I'm using for years. Hopefully, it will be possible :)
